From 483274b57697c66831cb5ac273df095f1e637219 Mon Sep 17 00:00:00 2001
From: Prithvi A Pai <prithvi.a.pai@intel.com>
Date: Wed, 6 Dec 2023 17:00:00 +0530
Subject: [PATCH] S3M Cert Chain verify by spdm - ignore DICE ext

Currently, OpenSSL doesnt support DICE attestation. Hence it is needed to
Ignore Critical extension.

Tested:
BMC as requester is able to verify the cert chain of S3M.
Hence the attestation is successful

Signed-off-by: Prithvi A Pai <prithvi.a.pai@intel.com>
---
 os_stub/cryptlib_openssl/pk/x509.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/os_stub/cryptlib_openssl/pk/x509.c b/os_stub/cryptlib_openssl/pk/x509.c
index 06ec1a2fc7..adfc8ae186 100644
--- a/os_stub/cryptlib_openssl/pk/x509.c
+++ b/os_stub/cryptlib_openssl/pk/x509.c
@@ -1877,6 +1877,9 @@ bool libspdm_x509_verify_cert(const uint8_t *cert, size_t cert_size,
     if (cert_store == NULL) {
         goto done;
     }
+    /* Dice certificate verification needs ignoring of critical extension. */
+    X509_STORE_set_flags(cert_store, X509_V_FLAG_IGNORE_CRITICAL);
+
     if (!(X509_STORE_add_cert(cert_store, x509_ca_cert))) {
         goto done;
     }
-- 
2.34.1

